### PR TITLE
Docs: Note dependency on promise-polyfill

### PR DESCRIPTION
**Pull request by stramel**
@arthurevans Pointed out that `promise-polyfill` needs to be a dependency of this now. I mimic'd how `iron-ajax` was handling this.

@rictic @justinfagnani @arthurevans PTAL :)

### CODE REVIEW

**Comment by arthurevans** [comment]
Meanwhile, Rob pointed out that promise is actually polyfilled automatically if you use webcomponents-lite or webcomponents-loader, which I was not aware of.  Yet, people were reporting the missing polyfill as an issue. I am more than a little confused here.

**Comment by stramel** [comment]
I'm there with ya @arthurevans... It turns out looking at the code that webcomponents-loader loads webcomponents-lite if promise is missing. Their polyfill uses es6-promise, which would be different than promise-polyfill? Seems a bit weird to me that lite includes that polyfill to me.

**Comment by justinfagnani** [comment]
We may need a @notwaldorf or @azakus to comment on what's up with the loader

**Comment by notwaldorf** [comment]
if you don't have `window.Promise`, the loader pretty much assumes you are IE11, and loads `webcomponents-lite` for you. Using a promise polyfill that doesn't set `window.Promise` sounds...improper? What's it even polyfilling?

**Comment by notwaldorf** [comment]
@stramel in your message i'm not sure I know who "their polyfill" is referring to -- who is the they? :)

The reason `lite` includes the polyfill is mostly historical. It's the "load this one thing and one thing only, and it will always work". That's how it worked in 0.7, and if we removed it a) everyone's apps would break by just updating and b) some people would complain for sure

**Comment by arthurevans** [comment]
@notwaldorf OK, here's the background:

I've been following another bug (https://github.com/PolymerElements/iron-ajax/issues/283) which reports problems running the iron-ajax demo on IE, allegedly because it uses Promise. And there's a note in the iron-ajax docs about how promise-polyfill has been moved out of dependencies, which suggests it... Might somehow be useful.

But if the Promise polyfill is indeed loaded by webcomponents-lite.js, this shouldn't be an issue.

It also makes me wonder why we _ever_ included the promise-polyfill in iron-ajax, but that is a separate issue. Main point is:

- If the Promise polyfill is part of webcomponentsjs, we shouldn't need a disclaimer on repos that use promise.
- The iron-ajax issue above probably ain't related to promises.

**Comment by stramel** [comment]
TL;DR I'm fine with whatever route you guys decide, just trying to do whatever it takes to get this ready by Polymer Summit 😬

@notwaldorf I was referring to "their" as `webcomponentsjs/webcomponents` which appears to be using `es6-promise`. 

v0.7 has been typically paired with 1.x components which in the case of `iron-ajax` was including `promise-polyfill`. Thus at least shipping both if on IE11. 


You guys have since decoupled the polyfill from `iron-ajax` and other elements using `Promise` in (2.0). But also, added `promise-polyfill` as a devDependency which means that, at least locally, there is 2 promise polyfills. 

I was informed part of the reason for decoupling the polyfill was because people wanted to pick their `Promise` polyfill. This seems to conflict with Webcomponentsjs decision to bundle it.

**Comment by notwaldorf** [comment]
> part of the reason for decoupling the polyfill was because people wanted to pick their Promise polyfill

This part is technically incorrect, which is the best part of incorrect 😅. The reason that we decoupled the polyfill is because polyfills should be pay to play. In the case of `iron-ajax#^1.0.0`, an app that only had to run on chrome (eg. chrome://settings) still had to download and use the Promise polyfill, even though it had a perfectly working native Promise implementation. It's also true that at that point since you, the user, are responsible for bringing a polyfill, you can choose which polyfill you want, but that's a far and away secondary goal imo.

**Comment by stramel** [comment]
@notwaldorf That makes perfect sense. 

What would you consider to be the correct way of noting this? And does the devDependency on `promise-polyfill` make sense?

**Comment by arthurevans** [comment]
OK, talked through this with @notwaldorf and I finally understand.

Promise is polyfilled in the v1 webcomponents polyfills—which makes sense because the v1 APIs use Promises (think customElements.whenDefined).

Promise is _not_ polyfilled in the v0 webcomponents polyfills. So if you're using the hybrid element with Polymer 1.x **you need to bring your own polyfill**.

If you're using the hybrid element with Polymer 2.x, **you don't need to bring your own promise polyfill, just use webcomponents-loader or webcomponents-lite**. Without which, it won't work on IE, anyway.

Hope that helps. It certainly helped me.

**Comment by stramel** [comment]
Loving this knowledge dump! 💪

Thank you all for the feedback and helping me understand it all better! I will be making a couple adjustments to help clarify the situation

**Comment by stramel** [comment]
@justinfagnani What do you think about merging this?

**Comment by justinfagnani** [comment]
Looks good!
